### DISCUSSION TITLE
ControlNET Batching Issues (Folder/Upload)

### DISCUSSION THREAD

**Discussion by SilverBullet126** [lllyasviel/stable-diffusion-webui-forge#260] [Q&A]
I was trying out Forge and found that ControlNET and using the batching options isn't working as expected. In A1111 or SD.Next, ControlNet would cycle through all the images in a folder and make the request batch numbers for each file. In Forge, it goes through the images, but only makes 4 images and then stops; but it does process each image (mapping). I've tried changing the batch count/size and that doesn't seem to make any difference.  

I've not changed/edited anything in the ControlNET settings and single image works as expected. 

Anyone had similar issue/experience? 

![2024-02-14_14-16-01](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/152666726/98cc73f3-4508-43bc-881b-ed170c5df344)

**Comment by ArnorWing** [comment]
I have same problem. Mine makes 1 picture and then preprocesses the folder but stops making more images .

**Reply by Boussouni** [reply]
idem for me, controlnet doesn't work on Forge SD

**Reply by Boussouni** [reply]
Maybe it's not adapted to XL Models ?

**Reply by VeteranXT** [reply]
It does not work at all. Applies 1st file and rest gives you poses.

**Reply by FLi79Za** [reply]
Same here. There is also no method to actually update ControlNet independently so I suspect fixes will take longer to implement compared to the Auto1111 extension which is updated almost daily.
